File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        for(int i = 0; i < matrix.size(); i++){
            int low = 0;
            int high = matrix[i].size() - 1;
            int mid;
            while(low <= high){
                mid = low + (high - low) / 2;
                if(matrix[i][mid] == target){
                    return true;
                }
                else if(matrix[i][mid] > target){
                    high = mid - 1;
                }
                else{
                    low = mid + 1;
                }
            }
        }
        return false;
    }
};

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        flag = False
        for arr in matrix:
            l = 0
            h = len(arr) - 1
            while l <= h:
                mid = (l + h) // 2
                if arr[mid] == target:
                    flag = True
                    return True
                elif arr[mid] > target:
                    h = mid - 1
                else:
                    l = mid + 1
        return flag
                
            